core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -175,6 +175,13 @@ class Partition(val topic: String,
       // add replicas that are new
       allReplicas.foreach(replica => getOrCreateReplica(replica))
       val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r)).toSet
+
+      // If the ISR size received from a LeaderAndIsrRequest is smaller than the existing ISR size,
+      // mark the isrShrinkRate meter. For example: This can happen if one of the followers is shutdown.
+      // Note that this metric is not triggered on follower -> leader transition
```

Comment:
Just to clarify: i.e. no cluster-wide metric apart from leader-election-rate, but there are no cluster-wide shrink rate metrics that would capture that the ISR shrunk due to leader failure